src/main/fc/boot.c

```diff
@@ -359,7 +362,8 @@ void init(void)
     const sonarHardware_t *sonarHardware = NULL;
     sonarGPIOConfig_t sonarGPIOConfig;
     if (feature(FEATURE_SONAR)) {
-        sonarHardware = sonarGetHardwareConfiguration(batteryConfig()->currentMeterType);
+        bool usingCurrentMeterIOPins = feature(FEATURE_CURRENT_METER) && batteryConfig()->currentMeterSource == CURRENT_METER_ADC;
+        sonarHardware = sonarGetHardwareConfiguration(usingCurrentMeterIOPins);
```

Comment:
https://github.com/cleanflight/cleanflight/pull/2396/files#diff-01552029522bcf9b845bcfaadb04e307R132

That's the reason why external function declarations in .c file are so evil ...  This one is quite benign, but it can be deadly ..